### PR TITLE
feat: use @googlemaps/js-api-loader

### PR DESCRIPTION
**Pull request by jpoehnelt**
Use [@googlemaps/js-api-loader](https://www.npmjs.com/package/@googlemaps/js-api-loader) to handle the dynamic script injection. This also exposes many additional options.

### CODE REVIEW

**Comment by jpoehnelt** [comment]
Looks like the CI is broken for forks.

**Comment by Tintef** [comment]
Hey @jpoehnelt, thanks for the PR. I'll review it over the weekend

File: src/GooglePlacesAutocomplete.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ const GooglePlacesAutocomplete: React.FC<GooglePlacesAutocompleteProps> = ({
   useEffect(() => {
     const init = async () => {
       try {
-        if (apiKey) await injectScript(apiKey, apiOptions);
+        await new Loader({apiKey, ...apiOptions}).load();
```
Comment (Tintef) [inline src/GooglePlacesAutocomplete.tsx:51]: What do you think about defaulting to only load the places library? 🤔

Comment (jpoehnelt) [reply src/GooglePlacesAutocomplete.tsx:51]: Probably makes sense to do that. I'll add the change.

Comment (jpoehnelt) [reply src/GooglePlacesAutocomplete.tsx:53]: done

Comment (Tintef) [reply src/GooglePlacesAutocomplete.tsx:53]: @jpoehnelt Why not set it as a default prop? 🤔

Comment (jpoehnelt) [reply src/GooglePlacesAutocomplete.tsx:53]: Do you mean, `apiOptions = {libraries: ["places"]},`? If so, user might be passing something like `apiOptions={{mapIds: ['asdf']}}` and are not concerned about the libraries being used.

Comment (Tintef) [reply src/GooglePlacesAutocomplete.tsx:53]: That's a good point 🤔  But what you are currently mutating a prop and that's not good.

Maybe you could do something like:
```
await new Loader({ apiKey, ...{ libraries: ['places'], ...apiOptions }).load()
```

wdyt?

**Comment by Tintef** [comment]
@jpoehnelt Published on version 3.2.2 🥳 Thanks for the contribution

**Pull request merged**